community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/idmapping/string/EncodingIdMapper.java

```diff
@@ -414,8 +415,14 @@ private int detectAndMarkCollisions( ProgressListener progress )
             progress.add( batch );
         }
         progress.done();
-        monitor.numberOfCollisions( numberOfCollisions );
-        return numberOfCollisions;
+
+        if ( numberOfCollisions > Integer.MAX_VALUE )
```

Comment:
Is this exception fatal? If it isn't, then if this counter overflowed once, it can overflow again, making it again positive and passing this check. That probably should not happen.